src/Commands/Connection/InfoCommand.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Pantheon\Terminus\Commands\Connection;
+
+use Consolidation\OutputFormatters\StructuredData\RowsOfFields;
+
+use Pantheon\Terminus\Commands\TerminusCommand;
+use Terminus\Collections\Sites;
+use Terminus\Models\Environment;
+
+/**
+ * Class InfoCommand
+ * @package Pantheon\Terminus\Commands\Connection
+ */
+class InfoCommand extends TerminusCommand
+{
+    protected $authorized = true;
+
+    /**
+     * Retrieve connection info for a specific environment such as git, sftp, mysql, redis
+     *
+     * @command connection:info
+     *
+     * @param string $site_name Name of the site to retrieve
+     * @param string $env_name  Name of the environement to retrieve: dev|test|live|branch
+     * @param string $filter    Parameter filter (optional)
+     * @param array  $options   [fields=<env,param,value>] [format=<table|csv|yaml|json>]
+     *
+     * @return RowsOfFields
+     *
+     * @field-labels
+     *   env: Environment
+     *   param: Parameter
+     *   value: Connection Info
+     *
+     * @usage connection:info awesome-site dev git_command --format=json
+     *   Display connection information only for the given parameter
+     *
+     */
+    public function connectionInfo(
+        $site_name,
+        $env_name,
+        $filter = null,
+        $options = ['format' => 'table', 'fields' => 'env,param,value']
+    ) {
+        $sites       = new Sites();
+        $site        = $sites->get($site_name);
+        $environment = $site->environments->get($env_name);
+
+        return new RowsOfFields($this->environmentParams($environment, $filter));
+    }
```

Comment:
@TeslaDethray @greg-1-anderson @ronan: Any opinions on what would constitute an effective test here?
Argument validation is effectively provided by `Symfony\Console`, `Site` and `Environment` accessors are (should be) exercised in the model units.

The only logic left then is asserting the `environmentParams()` collaborator is invoked with the correct arguments and that the return is a RowOfFields. Are either of those worthwhile?